src/components/daySelector/DaySelector.spec.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react-native';
+import DaySelector from './DaySelector';
+import { Day } from '../../models/app.model';
+
+describe('DaySelector', () => {
+  const daysOfWeek: Day[] = Object.values(Day);
+
+  it('renders days correctly', () => {
+    const { getByText } = render(
+      <DaySelector daysOfWeek={daysOfWeek} selectedDay={null} onSelectDay={() => {}} />
+    );
+
+    daysOfWeek.forEach((day) => {
+      const abbreviatedDay = day.substring(0, 3);
+      expect(getByText(abbreviatedDay)).toBeDefined();
```

Comment:
maybe add a test id and to the text element and then just expect each element to have the day in the formatted way? it looks like you're searching the whole DOM for a certain text instead of a specific element which is much more stable - I'll show you an example how this isn't tight enough:
remove line 15 and change `abbreviatedDay` to `day` and I think the test will still pass

Reply:
ok

Reply:
when i applied your suggestion, the test failed